src/plugins_exts.h

```diff
@@ -676,6 +684,99 @@ LIBYANG_API_DECL uint16_t *lyplg_ext_print_get_level(const struct lyspr_ctx *ctx
 LIBYANG_API_DECL void lyplg_ext_print_info_extension_instance(struct lyspr_ctx *ctx, const struct lysc_ext_instance *ext,
         ly_bool *flag);
 
+/*
+ * sprinter tree
+ */
+
+/**
+ * @brief Callback to print parent node of @p ext or to print the contents of the extension.
+ *
+ * Function is called in two different cases. If the printer_tree needs the tree-diagram form of a parent node,
+ * then @p flags and @p add_opts are set. In the second case, if printer_tree needs to print the contents of the
+ * extension, then @p flags and @p add_opts are set to NULL and function must prepare the nodes that should be
+ * printed using the lyplg_ext_sprinter_tree_* functions.
+ *
+ * @param[in] ext Extension instance.
+ * @param[out] ctx Context for the tree printer. Extension contents can be inserted into it by functions
+ * lyplg_ext_sprinter_ctree_add_ext_nodes(), lyplg_ext_sprinter_ctree_add_nodes() or by their ptree alternatives.
+ * @param[out] flags Override tree-diagram \<flags\> string in a parent node.
+ * @param[out] add_opts Additional tree-diagram \<opts\> string in a parent node which is printed before \<opts\>.
+ * @return LY_ERR value.
+ */
+typedef LY_ERR (*lyplg_ext_sprinter_ctree_clb)(struct lysc_ext_instance *ext, const struct lyspr_tree_ctx *ctx,
+        char **flags, char **add_opts);
```

Comment:
All the `char **` parameters should probably be `char * const*`, pointer to a const string.